src/structures/GroupChat.js

```diff
@@ -52,57 +53,65 @@ class GroupChat extends Chat {
     get participants() {
         return this.groupMetadata.participants;
     }
-
+    
     /**
-     * Adds a list of participants by ID to the group
-     * @param {Array<string>} participantIds 
-     * @returns {Promise<Object>}
+     * Internal function to change a number of participant's state..
+     * @param {string} type (promote|demote|add|remove) 
      */
-    async addParticipants(participantIds) {
-        return await this.client.pupPage.evaluate((chatId, participantIds) => {
+    async _changeParticipants(participantIds, type, sleep = null) {
+        return await this.client.pupPage.evaluate(async (chatId, participantIds, type, sleep) => {
+            if (type != 'add' && type != 'remove' && type != 'promote' && type != 'demote') return null;
             const chatWid = window.Store.WidFactory.createWid(chatId);
             const participantWids = participantIds.map(p => window.Store.WidFactory.createWid(p));
-            return window.Store.GroupParticipants.sendAddParticipants(chatWid, participantWids);
-        }, this.id._serialized, participantIds);
+            const status = [];
+            for (const participantWid of participantWids) {
+                status.push(await window.Store.GroupParticipants['send'+type.charAt(0).toUpperCase() + type.slice(1)+'Participants'](chatWid, [participantWid]));
+                if (sleep) {
+                    await Util.sleep(sleep);
+                }
+            }
+            return status;
+        }, this.id._serialized, participantIds, type, sleep);
+    }
+
+    /**
+     * Adds a list of participants by ID to the group
+     * @param {Array<string>} participantIds
+     * @param {?number} sleep optional, amount to sleep in milliseconds before adding the next participant
+     * @returns {Promise<Array<Object>>}
+     */
+    async addParticipants(participantIds, sleep = null) {
+        return this._changeParticipants(participantIds, 'add', sleep);
     }
 
     /**
      * Removes a list of participants by ID to the group
-     * @param {Array<string>} participantIds 
-     * @returns {Promise<Object>}
+     * @param {Array<string>} participantIds
+     * @param {?number} sleep optional, amount to sleep in milliseconds before removing the next participant
+     * @returns {Promise<Array<Object>>}
      */
-    async removeParticipants(participantIds) {
-        return await this.client.pupPage.evaluate((chatId, participantIds) => {
-            const chatWid = window.Store.WidFactory.createWid(chatId);
-            const participantWids = participantIds.map(p => window.Store.WidFactory.createWid(p));
-            return window.Store.GroupParticipants.sendRemoveParticipants(chatWid, participantWids);
-        }, this.id._serialized, participantIds);
+    async removeParticipants(participantIds, sleep = null) {
```

Comment:
Isn't better to have a sleep value by default?

Reply:
> Isn't better to have a sleep value by default?

ok, around 100ms will do